lib/azure/armrest/storage/disk_service.rb

```diff
@@ -11,6 +11,112 @@ class DiskService < ResourceGroupBasedService
         end
+
+        # Get the raw blob information for a managed disk. This is similar to
+        # the StorageAccount#get_blob_raw method, but applies only to a managed
+        # disk, whereas that method applies only to an individual storage
+        # account.
+        #
+        # As with the Storage#get_blob_raw method, you should pass a :range,
+        # :start_byte, :end_byte or :length option. If you want the entire
+        # image you must pass the :entire_image option, though this is generally
+        # not recommended. Unlike the Storage#get_blob_raw method, this method
+        # does not support the :date parameter.
+        #
+        # The +byte_options+ are as follows:
+        #
+        #   :range        => A range of bytes you want, e.g. 0..1023 to get first 1k bytes
+        #   :start_byte   => The starting byte number that you want to collect bytes for. Use
+        #                    this in conjunction with :length or :end_byte.
+        #   :end_byte     => The ending byte that you want to collect bytes for. Use this
+        #                    in conjunction with :start_byte.
+        #   :length       => If given a :start_byte, specifies the number of bytes from the
+        #                    the :start_byte that you wish to collect.
+        #   :entire_image => If set, returns the entire image in bytes. This will be a long
+        #                    running request that returns a large number of bytes.
+        #
+        # You may also pass a :durationInSeconds parameter, which indicates
+        # how long the privately generated SAS token should last. This token
+        # is used internally by requests that are used to access the requested
+        # information. By default it lasts for 1 hour.
+        #
+        # Get the information you need using:
+        #
+        # * response.body    - blob data (the raw bytes).
+        # * response.headers - blob metadata (a hash).
+        #
+        # Example:
+        #
+        #   vms = Azure::Armrest::VirtualMachineService.new(conf)
+        #   sds = Azure::Armrest::Storage::DiskService.new(conf)
+        #
+        #   vm = vms.get(vm_name, vm_resource_group)
+        #   os_disk = vm.properties.storage_profile.os_disk
+        #
+        #   disk_id = os_disk.managed_disk.id
+        #   disk = sds.get_by_id(disk_id)
+        #
+        #   # Get the first 1024 bytes
+        #   data = sds.get_blob_raw(disk.name, disk.resource_group, :range => 0..1023)
+        #
+        #   p data.headers
+        #   File.open('vm.vhd', 'a'){ |fh| fh.write(data.body) }
+        #
+        def get_blob_raw(disk_name, resource_group = configuration.resource_group, byte_options = {}, local_options = {})
+          validate_resource_group(resource_group)
+
+          local_options[:access] = 'read' # Must be 'read' at the moment
+          local_options[:durationInSeconds] ||= local_options[:duration_in_seconds] || 3600 # 1 hour default
```

Comment:
What are other available keys for `local_options`? Our users don't need to know how Azure API works; therefore we don't require them to provide the exact options (e.g., `durationInSeconds`) that accepted by the API. If there are no other required options I would have the method signature as
```
def get_blob_raw(disk_name, resource_group=..., options={})
```
The allowed keys in `options` will be the location parameters that you listed above, plus `:duration_in_seconds`